### PR TITLE
UCP/RKEY/TEST: Refactor rkey pack/unpack and tests

### PR DESCRIPTION
**Pull request by yosefe**
# Why
Preparation for passing system topology information in rkey buffer

### CODE REVIEW

File: src/ucp/core/ucp_rkey.inl [hunk 1]
```diff
@@ -11,6 +11,23 @@
...
+ucp_rkey_config_hash_func(ucp_rkey_config_key_t rkey_config_key)
+{
+    return (khint_t)rkey_config_key.md_map ^
+           (rkey_config_key.ep_cfg_index << 8) ^
+           (rkey_config_key.mem_type << 16);
+}
+
+static UCS_F_ALWAYS_INLINE int
+ucp_rkey_config_is_equal(ucp_rkey_config_key_t rkey_config_key1,
+                                ucp_rkey_config_key_t rkey_config_key2)
```
Comment (brminich) [inline src/ucp/core/ucp_rkey.inl:24]: indentation looks strange. is this Clang format?

File: src/ucp/core/ucp_worker.inl [hunk 2]
```diff
@@ -8,30 +8,39 @@
...
-ucp_worker_rkey_config_is_equal(ucp_rkey_config_key_t rkey_config_key1,
-                                ucp_rkey_config_key_t rkey_config_key2)
+/**
+ * Resolve remote key configuration key to a remote key configuration index
+ *
+ * @param [in]  worker          UCP worker to resolve configuration on
+ * @param [in]  key             Rkey configuration key
+ * @param [in]  lanes_distance  Distance of each lane to the remote memory
+ *                              buffer on the peer.
+ * @param [out] cfg_index_p     Filled with configuration index in the worker.
```
Comment (brminich) [inline src/ucp/core/ucp_worker.inl:29]: minor: please either add  `.` for each line in the end, or remove the existing ones for consistency

File: src/ucp/core/ucp_worker.inl [hunk 3]
```diff
@@ -8,30 +8,39 @@
...
 
-static UCS_F_ALWAYS_INLINE int
-ucp_worker_rkey_config_is_equal(ucp_rkey_config_key_t rkey_config_key1,
-                                ucp_rkey_config_key_t rkey_config_key2)
+/**
+ * Resolve remote key configuration key to a remote key configuration index
+ *
+ * @param [in]  worker          UCP worker to resolve configuration on
+ * @param [in]  key             Rkey configuration key
+ * @param [in]  lanes_distance  Distance of each lane to the remote memory
```
Comment (brminich) [inline src/ucp/core/ucp_worker.inl:27]: distance is missing in the params

File: src/ucs/type/serialize.h [hunk 4]
```diff
@@ -17,14 +17,29 @@
  *
  * @param _iter   Pointer to a pointer, representing the current element.
  * @param _type   Type of the current element.
+ * @param _size   Current element size.
```
Comment (brminich) [inline src/ucs/type/serialize.h:20]: is this an offset to advance the pointer? If this was a current element size, `ucs_serialize_next` would be enough, right?

Comment (yosefe) [reply src/ucs/type/serialize.h]: updated documentation

File: src/ucp/core/ucp_rkey.inl (hunk 1 above)
Comment (yosefe) [reply src/ucp/core/ucp_rkey.inl]: fixed

File: src/ucp/core/ucp_worker.inl (hunk 3 above)
Comment (yosefe) [reply src/ucp/core/ucp_worker.inl]: leftover; removed

**Review by brminich — APPROVED** [review]

**Review by dmitrygx — APPROVED** [review]

**Pull request merged**